app/metal-controller-manager/api/v1alpha1/server_types.go

```diff
@@ -34,11 +36,41 @@ type SystemInformation struct {
 	Family       string `json:"family,omitempty"`
 }
 
+func (a *SystemInformation) PartialEqual(b *SystemInformation) bool {
+	return partialEqual(a, b)
+}
+
 type CPUInformation struct {
 	Manufacturer string `json:"manufacturer,omitempty"`
 	Version      string `json:"version,omitempty"`
 }
 
+func (a *CPUInformation) PartialEqual(b *CPUInformation) bool {
+	return partialEqual(a, b)
+}
+
+func partialEqual(a, b interface{}) bool {
+	old := reflect.ValueOf(a)
+	new := reflect.ValueOf(b)
+
+	for i := 0; i < old.NumField(); i++ {
+		if old.Field(i).IsZero() {
+			// Skip zero values, since that indicates that the user did not supply
+			// the field, and does not want to compare it.
+			continue
+		}
```

Comment:
There is one edge case here. If a user specifies a completely empty qualifier in YAML (e.g. `cpu: {}`), then this will still match. I would argue that that is user error, but we _could_ build in a safeguard and require that at least one field isn't the zero value.